psdisk/psdisk.c

```diff
@@ -387,199 +338,188 @@ static int psdisk_updateImg(void)
+	LIST_INSERT_HEAD(&psdisk_common.list, node, ptrs);
+	psdisk_common.count++;
 
 	return 0;
 }
 
 
 static int psdisk_parseToSave(void)
 {
-	int i, ret = 0;
-	char *argsStr, *endptr;
+	struct plist_node_t *node = NULL;
 	const char *delimiter = ",";
-	struct plist_node_t *pNode = NULL;
-
-	const int MAX_PARTITION_OPTS = 4;
-	char *opts[MAX_PARTITION_OPTS];
+	char *args, *endptr;
+	char *opts[4];
+	int i, ret = 0;
 
 	/* Get data from optarg and separete them */
-	if ((argsStr = (char *)malloc(strlen(optarg) + 1)) == NULL) {
-		fprintf(stderr,"Cannot allocate memory, err: %s.\n", strerror(errno));
+	args = malloc(strlen(optarg) + 1);
+	if (args == NULL) {
+		fprintf(stderr, "Cannot allocate memory, err: %s.\n", strerror(errno));
 		return -1;
 	}
 
 	do {
-		strcpy(argsStr, optarg);
-		if ((opts[0] = strtok(argsStr, delimiter)) == NULL) {
-			fprintf(stderr,"Invalid number of arguments.\n");
+		strcpy(args, optarg);
+		if ((opts[0] = strtok(args, delimiter)) == NULL) {
+			fprintf(stderr, "Invalid number of arguments.\n");
 			break;
 		}
 
-		for (i = 1; i < MAX_PARTITION_OPTS; ++i) {
+		for (i = 1; i < sizeof(opts) / sizeof(opts[0]); i++) {
 			if ((opts[i] = strtok(NULL, delimiter)) == NULL) {
-				fprintf(stderr,"Invalid number of arguments.\n");
+				fprintf(stderr, "Invalid number of arguments.\n");
 				ret = 1;
 				break;
 			}
 		}
 
-		if (ret)
+		if (ret) {
 			break;
+		}
 
-		if ((pNode = (struct plist_node_t *)calloc(1, sizeof(struct plist_node_t))) == NULL) {
-			fprintf(stderr,"Cannot allocate memory, err: %s.\n", strerror(errno));
+		node = calloc(1, sizeof(struct plist_node_t));
+		if (node == NULL) {
+			fprintf(stderr, "Cannot allocate memory, err: %s.\n", strerror(errno));
 			break;
 		}
 
 		/* Parse partition name */
-		for (i = 0; i < strlen(opts[0]); ++i) {
+		for (i = 0; i < strlen(opts[0]); i++) {
 			if (!isalnum(opts[0][i])) {
-				fprintf(stderr,"Invalid partition name - %s.\n", opts[0]);
+				fprintf(stderr, "Invalid partition name - %s.\n", opts[0]);
 				ret = 1;
 				break;
 			}
 		}
 
-		if (ret)
+		if (ret) {
 			break;
+		}
 
-		if (strlen(opts[0]) >= sizeof(pNode->pHeader.name)) {
-			fprintf(stderr,"Invalid partition name - %s.\n", opts[0]);
+		if (strlen(opts[0]) >= sizeof(node->part.name)) {
+			fprintf(stderr, "Invalid partition name - %s.\n", opts[0]);
 			break;
 		}
 		else {
-			strncpy((char *)pNode->pHeader.name, opts[0], sizeof(pNode->pHeader.name));
+			strncpy((char *)node->part.name, opts[0], sizeof(node->part.name));
 		}
 
-		if (((pNode->pHeader.offset = strtoul(opts[1], &endptr, 0)) == 0 && strlen(endptr) != 0) || (pNode->pHeader.offset != 0 && strlen(endptr) != 0)) {
-			fprintf(stderr,"Partition %s has invalid offset %s.\n", pNode->pHeader.name, endptr);
+		if ((((node->part.offset = strtoul(opts[1], &endptr, 0)) == 0) && (strlen(endptr) != 0)) || ((node->part.offset != 0) && (strlen(endptr) != 0))) {
+			fprintf(stderr, "Partition %s has invalid offset %s.\n", node->part.name, endptr);
```

Comment:
```suggestion
		node->part.offset = strtoul(opts[1], &endptr, 0);
		if ((node->part.offset == 0) || (*endptr != '\0')) {
			fprintf(stderr, "Partition %s has invalid offset %s.\n", node->part.name, endptr);
```
And futher instead of `strlen(endptr) !=0`  the use of `*endptr != 0`.

Reply:
Done.